pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -246,198 +247,210 @@ func (tkn *SQLTokenizer) skipBlank() {
 	for unicode.IsSpace(tkn.lastChar) {
 		tkn.next()
 	}
+	tkn.grab()
+}
+
+// myFastToUpper returns a copy of the byte slice s with all Unicode letters mapped to
+// their upper case. It uses the array pointed to by dst as the destination if possible.
+func myFastToUpper(src, dst []byte) []byte {
+	dst = dst[:0]
+	isASCII, hasLower := true, false
+	for i := 0; i < len(src); i++ {
+		c := src[i]
+		if c >= utf8.RuneSelf {
+			isASCII = false
+			break
+		}
+		hasLower = hasLower || ('a' <= c && c <= 'z')
+	}
+	if cap(dst) < len(src) {
+		dst = make([]byte, 0, len(src))
+	}
+	if isASCII { // optimize for ASCII-only byte slices.
+		if !hasLower {
+			// Just return a copy.
+			return append(dst, src...)
```

Comment:
i think in this case can't we just use the original string?

Reply:
👍  Yes, we can.